src/integrations/admin/background-indexing-integration.php

```diff
@@ -147,13 +206,136 @@ public function register_shutdown_indexing() {
+
+	/**
+	 * Schedule background indexing every 15 minutes if the index isn't already up to date.
+	 *
+	 * @return void
+	 */
+	public function schedule_cron_indexing() {
+		if ( ! \wp_next_scheduled( 'wpseo_indexable_index_batch' ) && $this->should_index_on_cron() ) {
+			\wp_schedule_event( ( \time() + \HOUR_IN_SECONDS ), 'fifteen_minutes', 'wpseo_indexable_index_batch' );
+		}
+	}
+
+	/**
+	 * Limit cron indexing to 15 indexables per batch instead of 25.
+	 *
+	 * @param int $indexation_limit The current limit (filter input).
+	 *
+	 * @return int The new batch limit.
+	 */
+	public function throttle_cron_indexing( $indexation_limit ) {
+		if ( \wp_doing_cron() ) {
+			/**
+			 * Filter: Adds the possibility to limit the number of items that are indexed when in cron action.
+			 *
+			 * @api int $limit Maximum number of indexables to be indexed per indexing action.
+			 */
+			return \apply_filters( 'wpseo_cron_indexing_limit_size', 15 );
+		}
+
+		return $indexation_limit;
+	}
+
+	/**
+	 * Limit cron indexing to 3 links per batch instead of 5.
+	 *
+	 * @param int $link_indexation_limit The current limit (filter input).
+	 *
+	 * @return int The new batch limit.
+	 */
+	public function throttle_cron_link_indexing( $link_indexation_limit ) {
+		if ( \wp_doing_cron() ) {
+			/**
+			 * Filter: Adds the possibility to limit the number of links that are indexed when in cron action.
+			 *
+			 * @api int $limit Maximum number of link indexables to be indexed per link indexing action.
+			 */
+			return \apply_filters( 'wpseo_cron_link_indexing_limit_size', 3 );
+		}
+
+		return $link_indexation_limit;
+	}
+
+	/**
+	 * Determine whether cron indexation should be performed.
+	 *
+	 * @return bool Should cron indexation be performed.
+	 */
+	protected function should_index_on_cron() {
+		if ( ! $this->indexable_helper->should_index_indexables() ) {
+			return false;
+		}
+
+		// The filter supersedes everything when preventing cron indexation.
+		if ( apply_filters( 'Yoast\WP\SEO\enable_cron_indexing', true ) !== true ) {
+			return false;
+		}
+
+		return $this->indexing_helper->get_limited_filtered_unindexed_count_background( 1 ) > 0;
+	}
+
+	/**
+	 * Determine whether background indexation should be performed.
+	 *
+	 * @param int $shutdown_limit The shutdown limit used to determine whether indexation should be run.
+	 *
+	 * @return bool Should background indexation be performed.
+	 */
+	protected function should_index_on_shutdown( $shutdown_limit ) {
+		if ( ! $this->yoast_admin_and_dashboard_conditional->is_met() || ! $this->get_request_conditional->is_met() ) {
+			return false;
+		}
+
+		if ( ! $this->indexable_helper->should_index_indexables() ) {
+			return false;
+		}
+
+		$total_unindexed = $this->indexing_helper->get_limited_filtered_unindexed_count_background( $shutdown_limit );
+		if ( $total_unindexed === 0 || $total_unindexed > $shutdown_limit ) {
+			return false;
+		}
+
+		return ! $this->wp_cron_enabled_conditional->is_met();
```

Comment:
That's actually a good point moving that check at least above the `get_limited_filtered_unindexed_count_background()` one, so that we don't perform those queries needlessly (they *are* going to be performed anyway from the `should_index_on_cron()` one and we have transients around them to not be performed repeatedly but still, no need to even try and repeat them).

So, I did just that - but I don't think it makes much difference to put it first, it's equally important as a conditional as the other ones.